codegen/src/Azure.Iot.Operations.ProtocolCompilerLib/T4/communication/rust/Property/t4/RustPropertyConsumer.tt

```diff
@@ -0,0 +1,439 @@
+            return Err(<#=this.propertyName.GetTypeName(TargetLanguage.Rust, "write", "request", "builder", "error")#>::UninitializedField(
+                "maintainer_id",
+            ));
+        }
+
+<# } #>
+        self.inner_builder.topic_tokens(self.topic_tokens.clone());
+
+        self.inner_builder.build()
+    }
+}
+<# } #>
+
+/// Read requester for `<#=this.propertyName.AsGiven#>`
+pub struct <#=this.propertyName.GetTypeName(TargetLanguage.Rust, "read", "requester")#><C>(
+    rpc_command::Invoker<<#=this.serializerEmptyType.GetTypeName(TargetLanguage.Rust)#>, <#=this.readRespSchema.GetTypeName(TargetLanguage.Rust)#>, C>,
+)
+where
+    C: ManagedClient + Clone + Send + Sync + 'static,
+    C::PubReceiver: Send + Sync + 'static;
+
+impl<C> <#=this.propertyName.GetTypeName(TargetLanguage.Rust, "read", "requester")#><C>
+where
+    C: ManagedClient + Clone + Send + Sync + 'static,
+    C::PubReceiver: Send + Sync + 'static,
+{
+    /// Creates a new [`<#=this.propertyName.GetTypeName(TargetLanguage.Rust, "read", "requester")#>`]
+    ///
+    /// # Panics
+    /// If the DTDL that generated this code was invalid
+    pub fn new(application_context: ApplicationContext, client: C, options: &CommandInvokerOptions) -> Self {
+        let mut invoker_options_builder = rpc_command::invoker::OptionsBuilder::default();
+        if let Some(topic_namespace) = &options.topic_namespace {
+            invoker_options_builder.topic_namespace(topic_namespace.clone());
+        }
+
+        let mut topic_token_map: HashMap<String, String> = options
+            .topic_token_map
+            .clone()
+            .into_iter()
+            .map(|(k, v)| (format!("ex:{k}"), v))
+            .collect();
+
+        topic_token_map.insert("action".to_string(), "read".to_string());
+        topic_token_map.insert("modelId".to_string(), MODEL_ID.to_string());
+        topic_token_map.insert("consumerClientId".to_string(), client.client_id().to_string());
+<# if (!this.propertyName.IsEmpty) { #>
+        topic_token_map.insert("propertyName".to_string(), "<#=this.propertyName.AsGiven#>".to_string());
+<# } #>
+
+        let invoker_options = invoker_options_builder
+            .request_topic_pattern(PROPERTY_TOPIC_PATTERN)
+            .command_name("<#=this.readCommandName#>")
+            .topic_token_map(topic_token_map)
+            .response_topic_prefix(options.response_topic_prefix.clone())
+            .response_topic_suffix(options.response_topic_suffix.clone())
+            .build()
+            .expect("DTDL schema generated invalid arguments");
+
+        Self(
+            rpc_command::Invoker::new(application_context, client, invoker_options)
+                .expect("DTDL schema generated invalid arguments"),
+        )
+    }
+
+    /// Invokes the [`<#=this.propertyName.GetTypeName(TargetLanguage.Rust, "read", "request")#>`]
+    ///
+    /// # Errors
+    /// [`AIOProtocolError`] if there is a failure invoking the request
```

Comment:
Ideally this would detail some more of the custom errors, but I don't think that's new to this PR, nor am I sure that there's much that would be possible to generate here